docs/spec2006-tutorial.md

```diff
@@ -0,0 +1,517 @@
+
+### Building the Disk Image  
+Having prepared necessary scripts to create the disk image, in this step, we will build the disk image using [packer](https://www.packer.io/).  
+
+First, we download the packer binary.
+```sh
+cd disk-image/
+wget https://releases.hashicorp.com/packer/1.4.5/packer_1.4.5_linux_amd64.zip
+unzip packer_1.4.5_linux_amd64.zip
+```
+In launch_spec_experiments.py, we document how we obtain the binary as follows,  
+```python
+packer = Artifact.registerArtifact(
+    command = '''wget https://releases.hashicorp.com/packer/1.4.5/packer_1.4.5_linux_amd64.zip;
+    unzip packer_1.4.5_linux_amd64.zip;
+    ''',
+    typ = 'binary',
+    name = 'packer',
+    path =  'disk-image/packer',
+    cwd = 'disk-image',
+    documentation = 'Program to build disk images. Downloaded sometime in November from hashicorp.'
+)
+```
+Second, we create a packer script (a json file) that describes how the disk image will be built. 
+In this step, we assume that we have the SPEC 2006 ISO file in the disk-image/spec2006 folder. 
+In this script, the ISO file name is CPU2006v1.0.1.iso. 
+The script is available [here](https://github.com/darchr/gem5art/blob/master/docs/disks/spec2006/spec2006.json), and we save the file at disk-image/spec2006/spec2006.json.  
+
+In the root folder of experiment,
+```sh
+cd disk-image/spec2006/
+wget https://raw.githubusercontent.com/darchr/gem5art/master/docs/disks/spec2006/spec2006.json
+```
+To build the disk image,
+```sh
+cd disk-image/
+./packer validate spec2006/spec2006.json # validate the script, including checking the input files
+./packer build spec2006/spec2006.json
+```
+The process should not take more than 40 minutes on a fairly recent machine with a normal internet speed. 
+The disk image will be in disk-image/spec2006/spec2006-image/spec2006.  
+
+**Note:**: Packer will output a VNC port that could be used to inspect the building process. 
+Ubuntu has a built-in VNC viewer, namely Remmina.  
+
+**Note:**: [More about using packer and building disk images](disks.md).  
+
+Now, in launch_spec_experiments.py, we make an Artifact object of the disk image.  
+```python
+disk_image = Artifact.registerArtifact(
+    command = './packer build spec2006/spec2006.json',
+    typ = 'disk image',
+    name = 'spec2006',
+    cwd = 'disk-image/',
+    path = 'disk-image/spec2006/spec2006-image/spec2006',
+    inputs = [packer, experiments_repo, m5_binary,],
+    documentation = 'Ubuntu Server with SPEC 2006 installed, m5 binary installed and root auto login'
+)
+```
+### Compiling Linux Kernel  
+In this step, we will download Linux kernel source code and compile the Linux kernel. 
+The file of interest in this step is the vmlinux file.  
+
+First, we download the Linux kernel source code. 
+The following command will shallow clone the linux stable repository as well as checking out the tag v4.19.83, which contains the code for linux kernel version 4.19.83. 
+This also works well for other version numbers.  
```

Comment:
version 4.19.83 has been tested with gem5 as discussed in the other tutorial (put link here). We suggest using config files that have been tested with gem5.